extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java

```diff
@@ -749,6 +752,34 @@ static Match validateNestedExpressions(QuteConfig config, TemplateAnalysis templ
                     // Bean not found
                     return putResult(match, results, expression);
                 }
+            } else if (isDataNamespace(namespace) && !expression.getParts().isEmpty()) {
+                Expression.Part firstPart = expression.getParts().get(0);
+
+                String typeInfo = null;
+                if (checkedTemplate != null) {
```

Comment:
I think that this branch is unnecessary because `Template#getParameterDeclarations()` should contain the final set of param declarations...

Reply:
It didn't yesterday, I'll check again and get back to you.

Reply:
It's actually a non-sense because the "implicit" param declarations are only available during the build phase.

In any case, your code should probably first iterate over the "explicit" param declarations defined in the template and then try the `checkedTemplate.bindings`.

Reply:
I've switched the order as required and added validation for Message Bundle Data Namespace expressions (not sure how this will be taken ...).